ramhorns/src/template/mod.rs

```diff
@@ -37,14 +38,33 @@ pub struct Template<'tpl> {
 
     /// Source from which this template was parsed.
     source: Cow<'tpl, str>,
+
+    /// Partials used in this template
+    partials: Option<Partials<'tpl>>,
```

Comment:
We don't really need the entire `HashMap` at this point, we only care about having all the sources so that the compiler can make sure our `blocks` are always in memory, so a `Vec<Cow<'tpl, str>>` should be sufficient here. In theory, this could be rolled in with `source` to save us 3 words on the struct, at the cost of always doing an allocation, not sure that's better. Also reducing the `HashMap` to a `Vec` is extra work.